lib/samples/apply_scheduled_updates_to_preplanned_map_area/apply_scheduled_updates_to_preplanned_map_area_sample.dart

```diff
@@ -30,80 +31,163 @@ class ApplyScheduledUpdatesToPreplannedMapAreaSample extends StatefulWidget {
+              children: [
+                Expanded(
+                  child: ArcGISMapView(
+                    controllerProvider: () => _mapViewController,
+                    onMapViewReady: onMapViewReady,
                   ),
-                  Center(
-                    child: ElevatedButton(
-                      onPressed: _canUpdate ? syncUpdates : null,
-                      child: const Text('Update'),
+                ),
+                Row(
+                  mainAxisAlignment: MainAxisAlignment.spaceEvenly,
+                  crossAxisAlignment: CrossAxisAlignment.center,
+                  children: [
+                    Column(
+                      crossAxisAlignment: CrossAxisAlignment.start,
+                      mainAxisAlignment: MainAxisAlignment.spaceEvenly,
+                      children: [
+                        Text('Updates: ${_updateStatus.name.toUpperCase()}'),
+                        Text('Update Size: ${_updateSizeKB}KB'),
+                      ],
                     ),
-                  ),
-                ],
+                    Center(
+                      child: ElevatedButton(
+                        onPressed: _updating
+                            ? null
+                            : _canUpdate
+                                ? syncUpdates
+                                : resetMapPackage,
+                        child: _canUpdate
+                            ? const Text('Update')
+                            : const Text('Reset'),
+                      ),
+                    ),
+                  ],
+                )
+              ],
+            ),
+            // Display a progress indicator and prevent interaction before state is ready
+            Visibility(
+              visible: !_ready,
+              child: SizedBox.expand(
+                child: Container(
+                  color: Colors.white30,
+                  child: const Center(child: CircularProgressIndicator()),
+                ),
               ),
-            )
+            ),
           ],
         ),
       ),
     );
   }
 
   Future<void> onMapViewReady() async {
-    await prepareData();
-    _isInitialized = await loadMapPackageMap();
+    // Setting the path to the map package data
+    final appDir = await getApplicationDocumentsDirectory();
+    _dataUri = Uri.parse('${appDir.absolute.path}/canyonlands');
+
+    // Prepare (download and extract) the map package data
+    await _prepareData();
+
+    // Check if there is an update for the map package
+    await _checkForUpdates();
 
-    checkForUpdates();
+    setState(() => _ready = true);
   }
 
-  Future<void> prepareData() async {
-    await downloadSampleData(['740b663bff5e4198b9b6674af93f638a']);
-    final appDir = await getApplicationDocumentsDirectory();
-    _dataUri = Uri.parse('${appDir.absolute.path}/canyonlands');
+  // Reset the updated map pacakge to the original state
+  Future<void> resetMapPackage() async {
```

Comment:
I am following the declaration ordering we discussed in discussion #1905 in the devtopia repo. Properties, then public functions, then private functions. However, the properties are not in alphabetical order. I understand that in this case public and private don't really matter since there is just the one file. The public methods here are the functions called from the `build` method, like button handlers and the `onMapViewReady`. The private functions are workers that support the public functions.

Reply:
Ok, feel free to ignore.